### PR TITLE
Convert Review Category Report - Pagination

### PR DESCRIPTION
**Pull request by vvicario**
## Addresses
https://broadinstitute.atlassian.net/browse/BTRX-614

## Changes
Add pagination to review category report. 
## Testing
Go to Admin Menu => Review Category Report
---

- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Get a thumb from Belatrix
- [ ] **Submitter**: Get a thumb from @rushtong
- [ ] **Submitter**: Merge to develop using github's "Squash and Merge" feature
- [ ] **Submitter**: Delete branch after merge

### CODE REVIEW

File: grails-app/services/org/broadinstitute/orsp/QueryService.groovy [hunk 1]
```diff
@@ -1060,23 +1061,75 @@ class QueryService implements Status {
...
         SessionFactory sessionFactory = grailsApplication.getMainContext().getBean('sessionFactory')
         final session = sessionFactory.currentSession
-        final String query =
-                ' select * ' +
-                        ' from issue ' +
-                        ' where type = :projectType ' +
-                        ' and deleted = 0 '
-        final SQLQuery sqlQuery = session.createSQLQuery(query)
-        final results = sqlQuery.with {
+        final StringBuffer query = new StringBuffer(' select distinct i.id from issue i left outer join issue_extra_property ie on ie.issue_id = i.id ')
```
Comment (rushtong) [inline grails-app/services/org/broadinstitute/orsp/QueryService.groovy]: It seems to me that this query can be combined with the one below in `'select * from issue where id in :ids '` and get a distinct list of all columns in the `issue` table for the criteria, and ordered.

File: src/main/webapp/fundingSourceReport/FundingsSourceReport.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ import { spinnerService } from "../util/spinner-service";
 import { Spinner } from "../components/Spinner";
 import { TableComponent } from "../components/TableComponent";
 import { handleRedirectToProject, printData } from "../util/Utils";
-import { SORT_NAME_INDEX, STYLES } from "../util/FundingsSourceReportConstants";
+import { FUNDING_SORT_NAME_INDEX, STYLES_FUNDING_REPORTS } from "../util/ReportConstants";
```
Comment (rushtong) [inline src/main/webapp/fundingSourceReport/FundingsSourceReport.js]: Not sure on the exact syntax, but I think if you import as, you can revert a bunch of changes below that don't add any real information.
```suggestion
import { FUNDING_SORT_NAME_INDEX, STYLES_FUNDING_REPORTS as STYLES} from "../util/ReportConstants";
```

File: src/main/webapp/util/ReportConstants.js [hunk 3]
```diff
@@ -10,15 +10,26 @@ export const SORT_NAME_INDEX = {
...
   statusWidth: '98px',
   pisWidth: '80px',
   protocolWidth: '94px',
   fundingNameWidth: '124px',
-  generalWidth: '136px'
+  generalWidth: '136px',
 };
 
-
+export const STYLES_REVIEW_CATEGORIES = {
```
Comment (rushtong) [inline src/main/webapp/util/ReportConstants.js]: Another way of doing this so you don't have such long and unwieldy names, make one style object with sub objects:
```
export const styles = {
  fundingReport: { ... },
  reviewCategories: { ... }
}
```

File: src/main/webapp/reviewCategories/ReviewCategories.js [hunk 4]
```diff
@@ -1,34 +1,81 @@
...
+    formatter: (cell, row, rowIndex, colIndex) =>
+    div({},[
+      a({ href: handleRedirectToProject(component.serverURL, row.projectKey) },[row.projectKey])
+    ]),
+  },
+  {
+    dataField: 'summary',
+    text: 'Summary',
+    sort: true,
+    headerStyle: (column, colIndex) => {
```
Comment (tadeork) [inline src/main/webapp/reviewCategories/ReviewCategories.js]: It seems like `column, colIndex` values are not in use here. Are those default values of this type of functions?

File: src/main/webapp/reviewCategories/ReviewCategories.js [hunk 5]
```diff
@@ -38,90 +85,118 @@ class ReviewCategories extends Component {
 
   init = () => {
     spinnerService.showAll();
-    Report.getReviewCategory().then(resp => {
+    this.setState({ isAdmin: component.isAdmin });
```
Comment (tadeork) [inline src/main/webapp/reviewCategories/ReviewCategories.js]: It's better idea to update the state via function like: 
```suggestion
    this.setState(prev => { 
      prev.isAdmin = component.isAdmin;
      return prev;
     });
```

**Pull request merged**